array-manipulation-3/src/zip.js

```diff
@@ -0,0 +1,13 @@
+/* exported zip */
+function zip(first, second) {
+  const zipArray = [];
+  for (let i = 0; i < first.length; i++) {
+    if (first[i] && second[i]) {
```

Comment:
This is risky. If either of these is falsy (such as `0` or `false`, etc), your condition will not be met.

You can find the shortest length before looping.

```js
const length = Math.min(first.length, second.length);
```